File: sort-tool/go_sort_tool/sorter/sorter.go
package sorter

import (
	"bufio"
	"os"
	"sort"
	"strings"
)

func SortFile(filename string, unique bool) ([]string, error) {
    file, err := os.Open(filename)
    if err != nil {
        return nil, err
    }
    defer file.Close()

    var lines []string
    scanner := bufio.NewScanner(file)
    for scanner.Scan() {
        lines = append(lines, strings.ToUpper(strings.TrimSpace(scanner.Text())))
    }

    if err := scanner.Err(); err != nil {
        return nil, err
    }

    sort.Strings(lines)

    if unique {
        return removeDuplicates(lines), nil
    }

    return lines, nil
}

func removeDuplicates(lines []string) []string {
    var uniqueLines []string
    for i, line := range lines {
        if i == 0 || line != lines[i-1] {
            uniqueLines = append(uniqueLines, line)
        }
    }
    return uniqueLines
}

File: sort-tool/elixir_sort_tool/lib/elixir_sort/sorter.ex
defmodule ElixirSort.Sorter do
  def sort_file(filename, opts \\ []) do 
    unique = Keyword.get(opts, :unique, false)

    filename
    |> File.stream!()
    |> Enum.map(&String.trim/1)
    |> then(fn lines ->
      if unique do  
        Enum.uniq(lines)
      else
        lines
      end
    end)
    |> Enum.sort()
  end
end
